miden-lib/asm/sat/miden_wallet.masm

```diff
@@ -0,0 +1,56 @@
+use.miden::sat::account
+use.miden::sat::tx
+
+#! Add a single asset to an account's vault 
+#!
+#! Stack: [ASSET]
+#! Output: []
+#!
+#! - ASSET is the non-fungible asset of interest.
+export.receive_asset
+    exec.account::add_asset
+    dropw
+end
+
+#! Creates a note which sends a single asset to the specified recipient.
+#!
+#! Stack: [ASSET, tag, RECIPIENT]
+#! Output: []
+#!
+#! - ASSET is the non-fungible asset of interest.
+#! - tag is the tag to be included in the note.
+#! - RECIPIENT is the recipient of the note.
+export.send_asset
+    exec.account::remove_asset
+    exec.tx::create_note
+    dropw drop dropw
+end
+
+#! Authenticate a transaction. 
+#! WIP - FALCON VERIFICATION IS MISSING, NOT SURE IF HASHING IN CORRECT ORDER
+#! Stack: []
+#! Output: []
+#!
+#! - ASSET is the non-fungible asset of interest.
+#! - tag is the tag to be included in the note.
+#! - RECIPIENT is the recipient of the note.
+export.auth_tx
```

Comment:
As mentioned in https://github.com/0xPolygonMiden/miden-base/issues/22#issuecomment-1677883397 I would probably move `auth_tx` into a separate module.

Reply:
ok, let's do this